v2/spanner-to-sourcedb/terraform/samples/spanner-to-sharded-mysql/README.md

```diff
@@ -0,0 +1,508 @@
+       file, in the `reverse_replication_roles` resource.
+
+>    2. GCS service account - The list of roles can be found in the `main.tf` file,
+  in the `gcs_publisher_role` resource.
+
+[This](#adding-access-to-terraform-service-account) section in the FAQ
+provides instructions to add these roles to an existing service account.
+
+## Assumptions
+
+It takes the following assumptions -
+
+1. Ensure that the MySQL instance is correctly setup.  
+     1. Check that the MySQL credentials are correctly specified in the [source shards file](#sample-source-shards-file). 
+     2. Check that the MySQL server is up. 
+     3. The MySQL user configured in the [source shards file](#sample-source-shards-file) should have [INSERT](https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#priv_insert), [UPDATE](https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#priv_update) and [DELETE](https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#priv_delete) privileges on the database. 
+2. Ensure that the MySQL instance and Dataflow workers can establish connectivity with each other. Template automatically adds networking firewalls rules to enable this access. This can differ depending on the source configuration. Please validate the template rules and ensure that network connectivity can be established.
+3. The source instance with database containing reverse-replication compatible
+   schema is created.
+4. A session file has been generated to perform the spanner to source schema mapping.
+
+> **_NOTE:_**
+[SMT](https://googlecloudplatform.github.io/spanner-migration-tool/quickstart.html)
+> can be used to generate the session file.
+
+## Description
+
+This sample contains the following files -
+
+1. `main.tf` - This contains the Terraform resources which will be created.
+2. `outputs.tf` - This declares the outputs that will be output as part of
+   running this terraform example.
+3. `variables.tf` - This declares the input variables that are required to
+   configure the resources.
+4. `terraform.tf` - This contains the required providers and APIs/project
+   configurations for this sample.
+5. `terraform.tfvars` - This contains the dummy inputs that need to be populated
+   to run this example.
+6. `terraform_simple.tfvars` - This contains the minimal list of dummy inputs
+   that need to be populated to run this example.
+
+## Resources Created
+
+Given these assumptions, it uses a supplied source database connection
+configuration and creates the following resources -
+
+1. **Firewall rules** - These rules allow Dataflow VMs to connect to each other
+   and allow Dataflow VMs to connect to the source MySQL shards.
+2. **GCS buckets** - A GCS bucket to hold reverse replication metadata, such as
+   session and source shards files.
+3. **GCS objects** - Generates source shards configuration file to upload to
+   GCS.
+4. **Pubsub topic and subscription** - This contains GCS object notifications as
```

Comment:
Similarly here, with the gcs. based pipeline we should not be recommending pubsub.